### ISSUE TITLE
Accessibility: 

### ISSUE THREAD

**Issue by sugarmango66** [jupyterhub/outreachy#114]
### What page is this for?

https://ipython.org/donate.html

### WAVE accessibility report

My name is Suzan, one of outreachy applicants. I'm so glad to be introduced to and familiar with accessibility topics which I know little before through this project. I realize it's quite an important but often ignored aspect of  web applications. Especially for a person from a developing country where people with disabilities usually stay at home and tolerate limits and inconveniences of normal life for lack of public infrastructure and services. Maybe it's hard for me to build wheelchair ramps in real life but I feel honored if I can make a real difference by improving something in cyberspace. 
Below is the report of the WAVE tool which can help me analyze the errors and potential issues of the target webpage.

<img width="1552" alt="screenshot of wave report" src="https://user-images.githubusercontent.com/94340363/197331305-0978b605-8ea9-48f1-8c82-a0983ec705b4.png">

### The HTML element to be changed

```html
Issue identification:
(1) feedback type: errors; file: donate.html; code in line 6: 
<html xmlns="http://www.w3.org/1999/xhtml">

(2) feedback type: Contrast Errors
(2.1) file: _static/agogo.css; code in line 165(show 164-168 here): 
div.header div.rel a {
  color: #ce5c00;
  letter-spacing: .05em;
  font-weight: bold;
}
(2.2) file: _static/agogo.css; code in line 47(show 45-48 here):
/* Default body styles */
a {
  color: #ce5c00;
}
(2.3) file: donate.html; code in line 206:
<a href="#">Find out more...</a>
(2.4) file: donate.html; code in line 220:
<h1>Support IPython Development<a class="headerlink" href="#support-ipython-development" title="Permalink to this headline">¶</a></h1>
(2.5) file: donate.html; code in line 267-270:
    <div class="footer" role="contentinfo">
        &#169; Copyright the IPython development team.
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.8.5.
    </div>

(3) feedback type: Alerts
(3.1) file: donate.html; code in line 27:
<img class="logo" src="_static/IPy_header.png" alt="Logo"/>

(3.2) file: nbviewer_thm.js; code in line 7:
thm = '<div class="nbviewer-thm"> <a href="http://nbviewer.ipython.org"><img src="_static/nbviewer-thm-IMG.png" title="IPython Notebook Viewer"/></a></div>'.replace('IMG', ix); 
(3.3) file: donate.html; code in line 145-148:
<a href="books.html"><img
title="Packt Publishing donates a portion of the proceeds from this book to
support IPython's development." width="160" height="200" id="sidebar-book-cover"
style="margin-bottom: 10px;"></a>
```

### Your proposed HTML (or CSS), after change

```html
Fix suggestion:
(1) add lang attribute in html tag and set to responding language: 
<html lang="en" xmlns="http://www.w3.org/1999/xhtml">

(2.1) adjust the text color of div.rel a little deeper to satisfy 4.5:1 ratio:
  color: #ce4c00;
(2.2) adjust the text color of all anchor element by default a little deeper to satisfy 4.5:1 ratio:
  color: #ce4c00;
(2.3) comment the anchor element with dead link for now:
    <!-- <a href="#">Find out more...</a> -->
ps: First I try to adjust the color of text or background, then I notice the link is invalid. So I think we can comment it out to avoid confusion for the audience or add some real link content.
(2.4) This is interesting. Thanks to the contrast error warning, I noticed the presence of the symbol ¶ and now I know we use anchor tags to help locate a place like title inside the page. So I think we can just leave it alone since it's intended to hide.
(2.5) Though we can adjust the color of the footer in _static/agogo.css line 352, it's unnecessary to do that because info in this section doesn't matter a lot for the general audience. So I will skip the error.

(3.1) improve image alt attribute more informative:
            <img class="logo" src="_static/IPy_header.png" alt="logo of IPython shows interactive computing which links to its mainpage"/>
(3.2) add alt attribute to img to make it more friendly:
thm = '<div class="nbviewer-thm"> <a href="http://nbviewer.ipython.org"><img src="_static/nbviewer-thm-IMG.png" title="IPython Notebook Viewer" alt=”an example image which links to nbviewer.org website/”/></a></div>'.replace('IMG', ix); 
(3.3) add alt attribute to img to make it more accessible:
<a href="books.html"><img
title="Packt Publishing donates a portion of the proceeds from this book to
support IPython's development." alt=”a random book cover which links to book collection page” width="160" height="200" id="sidebar-book-cover"
style="margin-bottom: 10px;"></a>
```

### More info

In addition to the WAVE report guidance, I notice something to be improved with this webpage. Below are what I spot and my suggestions.
(1) file: donate.html; code in line 86:
`  <img src="_static/jupytercon-logo.svg" width="180px" alt="JupyterCon 2017">
`
Suggestion: update the year in alt text.
After change:  `<img src="_static/jupytercon-logo.svg" width="180px" alt="JupyterCon 2023">`

(2) file: donate.html; code in line 50:
`    <div id="cse-search-form" style="width: 100%;">Loading</div>
`
Suggestion: change the div text to be more descriptive and avoid confusing words.
After change:     `<div id="cse-search-form" style="width: 100%;">Useful links</div>`

After all the effort to the microtask, I have to say it's more demanding than I expected. But it's truly rewarding and opens a door to a new world for me. A lot of thanks, jupyterhub community!

**Comment by sugarmango66** [comment]
Hi mentors @sgibson91 @minrk @GeorgianaElena , lucky me to have the opportunity to make a contribution to jupyterhub. Thank you for your giving on helping open source newcomers like me start a wonderful journey! To be honest it's my first time to open an issue. I get stuck now and then but I'm happy I made it at last. The  process really benefits me a lot. Please help review my submission if convenient and I hope to receive your feedback and guidance.
Thousands of Thanks.

**Comment by minrk** [comment]
Thanks for the suggestions!

**Comment by sgibson91** [comment]
Thank you for your contribution! Since the contribution period is now over, we will close this issue.
